### PR TITLE
Migrate scratch-gui to GitHub Actions

### PR DESCRIPTION
**Pull request by delasare**
### Resolves
https://scratchfoundation.atlassian.net/browse/ENG-11

- Adds nvmrc
- Adds github actions workflow file for scratch-gui
- Adds github actions translations daily job

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
     "build": "npm run clean && webpack --colors --bail",
     "clean": "rimraf ./build && mkdirp build && rimraf ./dist && mkdirp dist",
     "deploy": "touch build/.nojekyll && gh-pages -t -d build -m \"[skip ci] Build for $(git log --pretty=format:%H -n1)\"",
-    "prepare": "husky install",
```
Comment (AlexisGoodfellow) [inline package.json:17]: Good riddance!

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            DETECT_CHROMEDRIVER_VERSION: "true"
+            JEST_JUNIT_OUTPUT_DIR: test-results
+            NODE_OPTIONS: --max-old-space-size=4000
+        steps:
+            - uses: actions/checkout@v4
+            - uses: wagoid/commitlint-github-action@v5
+              if: github.ref_name == 'develop' 
+            - uses: actions/setup-node@v3
+              with:
+                  cache: "npm"
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml:30]: I believe this line tells the `setup-node` action to cache the npm package cache, so the "Cache node_modules" item further down shouldn't be necessary

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            - name: Info
+              run: |
+                  cat <<EOF
+                  Node version: $(node --version)
+                  NPM version: $(npm --version)
+                  GitHub ref: ${{ github.ref }}
+                  GitHub head ref: ${{ github.head_ref }}
+                  EOF
+            - name: Install Dependencies
+              run: npm install
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml]:
```suggestion
              run: npm ci
```

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (cwillisf) [reply .github/workflows/ci-cd.yml:30]: Oh, never mind. I see now that you're using those cache items to carry files to the other jobs.

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                    src/generated
+                key: ${{ runner.os }}-generated-${{ hashFiles('package-lock.json') }}
+            - name: Retireve static/microbit
+              uses: actions/cache@v3
+              with:
+                path: 
+                    static/microbit
+                key: ${{ runner.os }}-microbit-${{ hashFiles('package-lock.json') }}
+            - name: Run Build
+              run: npm run build
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml]:
```suggestion
            - name: Run Build
              env:
                NODE_ENV: production
              run: npm run build
```

File: .github/workflows/ci-cd.yml [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+                path: 
+                    ./build
+                key: ${{ runner.os }}-build-${{ hashFiles('package-lock.json') }}
+            - uses: browser-actions/setup-chrome@v1
+              with:
+                chrome-version: stable
+              id: setup-chrome
+            - run: |
+                echo Installed chromium version: ${{ steps.setup-chrome.outputs.chrome-version }}
+                ${{ steps.setup-chrome.outputs.chrome-path }} --version
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml:178]: :sunglasses:

**Review by cwillisf — CHANGES_REQUESTED** [review]
I noticed a couple small changes that I think are needed to keep consistent with previous behavior, but it's looking good!

**Review by cwillisf — APPROVED** [review]
:+1: :rocket:

**Pull request merged**